rules/no-instanceof-builtin-object.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+const {checkVueTemplate} = require('./utils/rule.js');
+const {getParenthesizedRange} = require('./utils/parentheses.js');
+const {replaceNodeOrTokenAndSpacesBefore, fixSpaceAroundKeyword} = require('./fix/index.js');
+
+const isInstanceofToken = token => token.value === 'instanceof' && token.type === 'Keyword';
+
+const MESSAGE_ID = 'no-instanceof-builtin-object';
+const messages = {
+	[MESSAGE_ID]: 'Unsafe instanceof should not be used to check type.',
+};
+
+const primitiveConstructors = new Set(['String', 'Number', 'Boolean', 'BigInt', 'Symbol']);
+
+const referenceConstructors = new Set([
+	// Error types
+	'AggregateError',
+	'Error',
+	'EvalError',
+	'RangeError',
+	'ReferenceError',
+	'SyntaxError',
+	'TypeError',
+	'URIError',
```

Comment:
Please extract them and share with https://github.com/sindresorhus/eslint-plugin-unicorn/blob/4e539b4b084ce06165a0fdbc02d309ce7cf75c82/rules/error-message.js#L15